sdk/eventhub/azure-eventhub/azure/eventhub/_producer.py

```diff
@@ -112,72 +119,41 @@ def __init__(self, client, target, **kwargs):
         self._handler = None  # type: Optional[SendClient]
-        self._outcome = None  # type: Optional[constants.MessageSendResult]
         self._condition = None  # type: Optional[Exception]
         self._lock = threading.Lock()
-        self._link_properties = {
-            types.AMQPSymbol(TIMEOUT_SYMBOL): types.AMQPLong(int(self._timeout * 1000))
-        }
+        self._link_properties = {TIMEOUT_SYMBOL: pyamqp_utils.amqp_long_value(int(self._timeout * 1000))}
 
     def _create_handler(self, auth):
         # type: (JWTTokenAuth) -> None
-        self._handler = PySendClient(
+        self._handler = SendClient(
             self._client._address.hostname,
             self._target,
             auth=auth,
-            idle_timeout=10,
-            network_trace=self._client._config.network_tracing
+            idle_timeout=self._idle_timeout,
+            network_trace=self._client._config.network_tracing,
+            retry_policy=self._retry_policy,
+            keep_alive_interval=self._keep_alive,
+            client_name=self._name,
+            link_properties=self._link_properties,
+            properties=create_properties(self._client._config.user_agent),  # pylint: disable=protected-access
         )
 
     def _open_with_retry(self):
         # type: () -> None
         return self._do_retryable_operation(self._open, operation_need_param=False)
 
-    def _set_msg_timeout(self, timeout_time, last_exception):
-        # type: (Optional[float], Optional[Exception]) -> None
-        if not timeout_time:
-            return
-        remaining_time = timeout_time - time.time()
-        if remaining_time <= 0.0:
-            if last_exception:
-                error = last_exception
-            else:
-                error = OperationTimeoutError("Send operation timed out")
-            _LOGGER.info("%r send operation timed out. (%r)", self._name, error)
-            raise error
-        self._handler._msg_timeout = remaining_time * 1000  # type: ignore  # pylint: disable=protected-access
-
     def _send_event_data(self, timeout_time=None, last_exception=None):
         # type: (Optional[float], Optional[Exception]) -> None
         if self._unsent_events:
             self._open()
-            self._set_msg_timeout(timeout_time, last_exception)
-            self._handler.send_message(self._unsent_events[0])
+            timeout = timeout_time - time.time() if timeout_time else 0
+            self._handler.send_message(self._unsent_events[0], timeout=timeout)
             self._unsent_events = None
-            # self._unsent_events = self._handler.pending_messages  # type: ignore
-            # if self._outcome != constants.MessageSendResult.Ok:
-            #     if self._outcome == constants.MessageSendResult.Timeout:
-            #         self._condition = OperationTimeoutError("Send operation timed out")
-            #     if self._condition:
-            #         raise self._condition
```

Comment:
We no longer need to monitor unsent messages?

Reply:
We need to monitor because amqp send client `queue` and `wait` were used to send messages -- We didn't use uamqp send_message because there was an issue with the implementation (probably dealing with timeout?).

but now we have reimplemented the lib, I think it's time to use the `send_message` method which is simpler and the right way to do it